### PR TITLE
Fix unstabilizing comments on assignments

### PR DESCRIPTION
**Pull request by emillon**
Just a test case for #1088 for now.

### CODE REVIEW

**Comment by gpetiot** [comment]
No diff with test_branch (default profile and janestreet profile), no issue with the `<-` operator.

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -1458,10 +1453,19 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       , [(Nolabel, r); (Nolabel, v)] )
     when is_simple c.conf width (sub_exp ~ctx r) ->
       Cmts.relocate c.cmts ~src:pexp_loc ~before:loc ~after:loc ;
+      let break_before, break_after =
+        match c.conf.assignment_operator with
+        | `Begin_line -> (break 1 2, char ' ')
+        | `End_line -> (char ' ', break 1 2)
+      in
```
Comment (Julow) [inline src/Fmt_ast.ml]: This part is unnecessary. It's the same as before.

File: test/passing/assign_comment.ml.ref [hunk 2]
```diff
@@ -0,0 +1,35 @@
+(* see https://github.com/ocaml-ppx/ocamlformat/issues/1088 *)
+
+let _ =
+  r
+    (* _________________________________________________________________ *) :=
```
Comment (Julow) [inline test/passing/assign_comment.ml.ref]: This exceed the margin (78/77)

Comment (gpetiot) [reply test/passing/assign_comment.ml.ref]: fixed

**Comment by emillon** [comment]
When I started this PR I didn't know about `js_source`, maybe the test file can be merged there? I like small test cases but since all JS examples are already in a single place, it makes sense to keep it there.

In any case, this can be merged I think.

**Comment by gpetiot** [comment]
> When I started this PR I didn't know about `js_source`, maybe the test file can be merged there? I like small test cases but since all JS examples are already in a single place, it makes sense to keep it there.
> 
> In any case, this can be merged I think.

I think it's better here to have a smaller file so we can test both values of the option on a small code snippet.

File: test/passing/assign_comment-begin_line.ml.ref [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+let _ =
+  r
+    (* _________________________________________________________________ *)
+    (* _________________________________________________________________ *)
+    := 1
+
+let _ =
+  r
+    := (* _________________________________________________________________ *)
```
Comment (Julow) [inline test/passing/assign_comment-begin_line.ml.ref]: This comment exceed the margin by 1. Breaking the line won't help so maybe it's ok ?

File: src/Fmt_ast.ml [hunk 4]
```diff
@@ -1466,11 +1461,23 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       , [(Nolabel, r); (Nolabel, v)] )
     when is_simple c.conf (expression_width c) (sub_exp ~ctx r) ->
       Cmts.relocate c.cmts ~src:pexp_loc ~before:loc ~after:loc ;
+      let adj = fmt_if Poly.(c.conf.assignment_operator = `End_line) "@," in
```
Comment (Julow) [inline src/Fmt_ast.ml]: Should move this line inside the `cmts_before` let.

File: src/Fmt_ast.ml [hunk 5]
```diff
@@ -1466,11 +1461,23 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
...
-           $ Cmts.fmt c loc (fmt_assign_colon c)
-           $ hvbox 2 (fmt_expression c (sub_exp ~ctx v)) ))
+           ( match c.conf.assignment_operator with
+           | `Begin_line ->
+               hvbox 0 (fmt_expression c (sub_exp ~ctx r) $ cmts_before)
+               $ fmt "@;<1 2>:= " $ cmts_after
+               $ hvbox 2 (fmt_expression c (sub_exp ~ctx v))
+           | `End_line ->
+               hvbox 0
+                 (fmt_expression c (sub_exp ~ctx r) $ cmts_before $ fmt " :=")
```
Comment (Julow) [inline src/Fmt_ast.ml]:
```suggestion
                 (fmt_expression c (sub_exp ~ctx r) $ cmts_before $ str " :=")
```

File: test/passing/assign_comment-begin_line.ml.ref (hunk 3 above)
Comment (gpetiot) [reply test/passing/assign_comment-begin_line.ml.ref]: Yes there is already enough breaks before and after the comment, at some point the user needs to put spaces in comments :D

**Comment by gpetiot** [comment]
I reorganized the tests, I agree `Pexp_setfield` and `Pexp_setinstvar` could be handled in a separate PR.

**Pull request merged**